### PR TITLE
Fixed agency access

### PR DESCRIPTION
**Pull request by ewoutp**
This PR solves the problem that the starter could not talk the the agency (in 100% of the cases) when authentication was turned on.

### CODE REVIEW

File: service/peer.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func (p Peer) CreateDBServerAPI(clientBuilder ClientBuilder) (driver.Client, err
 		port := p.Port + p.PortOffset + ServerType(ServerTypeDBServer).PortOffset()
 		scheme := NewURLSchemes(p.IsSecure).Browser
 		ep := fmt.Sprintf("%s://%s", scheme, net.JoinHostPort(p.Address, strconv.Itoa(port)))
-		c, err := clientBuilder([]string{ep}, false)
+		c, err := clientBuilder([]string{ep}, ConnectionTypeDatabase)
```
Comment (mchacki) [inline service/peer.go]: This now has a different value for DontFollowRedirects is this correct?

File: service/peer.go [hunk 2]
```diff
@@ -117,7 +117,7 @@ func (p Peer) CreateCoordinatorAPI(clientBuilder ClientBuilder) (driver.Client,
 		port := p.Port + p.PortOffset + ServerType(ServerTypeCoordinator).PortOffset()
 		scheme := NewURLSchemes(p.IsSecure).Browser
 		ep := fmt.Sprintf("%s://%s", scheme, net.JoinHostPort(p.Address, strconv.Itoa(port)))
-		c, err := clientBuilder([]string{ep}, false)
+		c, err := clientBuilder([]string{ep}, ConnectionTypeDatabase)
```
Comment (mchacki) [inline service/peer.go]: This now has a different value for DontFollowRedirects is this correct?

File: service/upgrade_manager.go [hunk 3]
```diff
@@ -392,7 +392,7 @@ func (m *upgradeManager) areDBServersResponding(ctx context.Context) error {
 	}
 	// Check all
 	for _, ep := range endpoints {
-		c, err := m.upgradeManagerContext.CreateClient([]string{ep}, false)
+		c, err := m.upgradeManagerContext.CreateClient([]string{ep}, ConnectionTypeDatabase)
```
Comment (mchacki) [inline service/upgrade_manager.go]: This now has a different value for DontFollowRedirects is this correct?

File: service/upgrade_manager.go [hunk 4]
```diff
@@ -414,7 +414,7 @@ func (m *upgradeManager) areCoordinatorsResponding(ctx context.Context) error {
...
```
Comment (mchacki) [inline service/upgrade_manager.go]: This now has a different value for DontFollowRedirects is this correct?

File: service/upgrade_manager.go [hunk 5]
```diff
@@ -436,7 +436,7 @@ func (m *upgradeManager) areSingleServersResponding(ctx context.Context) error {
...
```
Comment (mchacki) [inline service/upgrade_manager.go]: This now has a different value for DontFollowRedirects is this correct?

File: service/upgrade_manager.go [hunk 6]
```diff
@@ -546,7 +546,7 @@ func (m *upgradeManager) ShowArangodServerVersions(ctx context.Context) (bool, e
 	showGroup := func(serverType ServerType, endpoints []string) {
 		groupVersions := make([]string, len(endpoints))
 		for i, ep := range endpoints {
-			c, err := m.upgradeManagerContext.CreateClient([]string{ep}, false)
+			c, err := m.upgradeManagerContext.CreateClient([]string{ep}, ConnectionTypeDatabase)
```
Comment (mchacki) [inline service/upgrade_manager.go]: This now has a different value for DontFollowRedirects is this correct?

**Comment by ewoutp** [comment]
The answer is yes to all, allthough I'll do 1 more check for activefailover.

For everything except agents, the follow-redirect is irrelevant because the servers will not yield a 307.

**Comment by mchacki** [comment]
i think you are right. 307  is never created by DBServers.

**Pull request merged**